app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -21,31 +21,42 @@ class IntentReceiverActivity : Activity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        // The intent property is nullable, but the rest of the code below
-        // assumes it is not. If it's null, then we make a new one and open
-        // the HomeActivity.
-        val intent = intent?.let { Intent(intent) } ?: Intent()
-
         val isPrivate = Settings.getInstance(this).usePrivateMode
 
-        if (isPrivate) {
-            components.utils.privateIntentProcessor.process(intent)
-        } else {
-            components.utils.intentProcessor.process(intent)
+        MainScope().launch {
+            // The intent property is nullable, but the rest of the code below
+            // assumes it is not. If it's null, then we make a new one and open
+            // the HomeActivity.
+            val intent = intent?.let { Intent(intent) } ?: Intent()
+
+            val intentProcessors = listOf(
+                components.utils.customTabIntentProcessor,
+                if (isPrivate) components.utils.privateIntentProcessor else components.utils.intentProcessor
+            )
+
+            intentProcessors.any { it.process(intent) }
+            setIntentActivity(intent)
+
+            startActivity(intent)
+
+            finish()
         }
+    }
 
+    private fun setIntentActivity(intent: Intent) {
         val openToBrowser = when {
-            CustomTabConfig.isCustomTabIntent(SafeIntent(intent)) -> {
-                intent.setClassName(
-                    applicationContext,
-                    if (intent.hasExtra(getString(R.string.intent_extra_auth))) AuthCustomTabActivity::class.java.name
-                    else CustomTabActivity::class.java.name
-                )
+            components.utils.customTabIntentProcessor.matches(intent) -> {
+                val activityClass = if (intent.hasExtra(getString(R.string.intent_extra_auth))) {
```

Comment:
I have no idea why the key to an extra is a `@StringRes`.

Reply:
Should we move it to be a constant?